write-fonts/src/tables/mvar.rs

```diff
@@ -0,0 +1,124 @@
+        [
+            VariationRegion::new(vec![reg_coords(0.0, 1.0, 1.0)]),
+            VariationRegion::new(vec![reg_coords(0.0, 0.5, 1.0)]),
+            VariationRegion::new(vec![reg_coords(0.5, 1.0, 1.0)]),
+        ]
+    }
+
+    fn read_metric_delta(mvar: &read_mvar::Mvar, tag: &[u8; 4], coords: &[f32]) -> f64 {
+        let coords = coords
+            .iter()
+            .map(|c| F2Dot14::from_f32(*c))
+            .collect::<Vec<_>>();
+        mvar.metric_delta(Tag::new(tag), &coords).unwrap().to_f64()
+    }
+
+    #[test]
+    fn simple_smoke_test() {
+        let [r1, r2, r3] = test_regions();
+        let mut builder = VariationStoreBuilder::new();
+        let delta_ids = vec![
+            // deltas for horizontal ascender 'hasc' only defined for 1 region
+            builder.add_deltas(vec![(r1, 10)]),
+            // deltas for horizontal descender 'hdsc' defined for 2 regions
+            builder.add_deltas(vec![(r2, -20), (r3, -30)]),
+        ];
+        let (varstore, index_map) = builder.build();
+
+        let mut value_records = Vec::new();
+        for (tag, temp_id) in [b"hasc", b"hdsc"].into_iter().zip(delta_ids.into_iter()) {
+            let varidx = index_map.get(temp_id).unwrap();
+            let value_record = ValueRecord::new(
+                Tag::new(tag),
+                varidx.delta_set_outer_index,
+                varidx.delta_set_inner_index,
+            );
+            value_records.push(value_record);
+        }
+
+        let table = Mvar::new(MajorMinor::new(1, 0), Some(varstore), value_records);
+
+        let bytes = dump_table(&table).unwrap();
+        let read = read_mvar::Mvar::read(FontData::new(&bytes)).unwrap();
+
+        assert_eq!(read.version(), table.version);
+        assert_eq!(read.value_record_count(), 2);
+        assert_eq!(read.value_record_size(), 8);
+        assert!(read.item_variation_store().is_some());
+
+        assert_eq!(read.value_records()[0].value_tag(), Tag::new(b"hasc"));
+        assert_eq!(read.value_records()[0].delta_set_outer_index(), 0);
+        assert_eq!(read.value_records()[0].delta_set_inner_index(), 1);
+        assert_eq!(read_metric_delta(&read, b"hasc", &[0.0]), 0.0);
+        // at axis coord 0.5, the interpolated delta will be half of r1's delta
+        assert_eq!(read_metric_delta(&read, b"hasc", &[0.5]), 5.0);
+        assert_eq!(read_metric_delta(&read, b"hasc", &[1.0]), 10.0);
+
+        assert_eq!(read.value_records()[1].value_tag(), Tag::new(b"hdsc"));
+        assert_eq!(read.value_records()[1].delta_set_outer_index(), 0);
+        assert_eq!(read.value_records()[1].delta_set_inner_index(), 0);
+        assert_eq!(read_metric_delta(&read, b"hdsc", &[0.0]), 0.0);
+        // this coincides with the peak of intermediate region r2, hence != 30.0/2
+        assert_eq!(read_metric_delta(&read, b"hdsc", &[0.5]), -20.0);
+        assert_eq!(read_metric_delta(&read, b"hdsc", &[1.0]), -30.0);
+    }
```

Comment:
Optional, IMHO a helper or reading into a struct so you could have coarser asserts would make this easier to read. For example, `assert_value_record(what it should look like, value_record[0]);`

I find it easier to troubleshoot because when it fails I can see all the values that are wrong not just the first one.